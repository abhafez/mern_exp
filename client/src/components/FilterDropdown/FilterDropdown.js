import React from 'react';
import PropTypes from 'prop-types';
import { Dropdown } from 'semantic-ui-react';

const FilterDropdown = ({ value, options, onFilter }) => (
  <Dropdown
    options={options}
    search
    value={value}
    selection
    onChange={onFilter}
    testid="dropdown"
  />
);

FilterDropdown.propTypes = {
  value: PropTypes.string,
  options: PropTypes.array,
  onFilter: PropTypes.func,
};

export default FilterDropdown;
