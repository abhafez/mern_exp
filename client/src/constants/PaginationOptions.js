export const options = [
  { key: 1, text: '5 Items', value: 5 },
  { key: 2, text: '8 Items', value: 8 },
  { key: 3, text: '12 Items', value: 12 },
];
