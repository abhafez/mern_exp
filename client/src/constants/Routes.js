/**
 * @constant {Object} Routes
 * @description List of websites routes
 * Accessibile by importing 'Routes' in any place but './Routes' in the main index.js
 * You should not need to import it there
 */
const Routes = {
  home: '/',
  login: '/login',
  LoadingIndicator: '/LoadingIndicator',
  testing: '/testing',
};

module.exports = {
  Routes,
};
