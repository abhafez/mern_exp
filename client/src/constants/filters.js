export const departments = [
  { key: 1, text: 'Gaming', value: 'Gaming' },
  { key: 2, text: 'Toys', value: 'Toys' },
  { key: 3, text: 'Gifts', value: 'Gifts' },
  { key: 4, text: 'Books', value: 'Books' },
];

export const promos = [
  { key: 1, text: 'Toy-20', value: 'Toy-20' },
  { key: 2, text: 'Game-30', value: 'Game-30' },
  { key: 3, text: 'Gift-50', value: 'Gift-50' },
  { key: 4, text: 'Book-80', value: 'Book-80' },
];
