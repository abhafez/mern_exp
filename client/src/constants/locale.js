/**
 * set the project's default locale here
 */

export const DEFAULT_LOCALE = localStorage.getItem('locale') || 'en';
