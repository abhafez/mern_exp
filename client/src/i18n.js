/** i18n setup */

/**
 * Getting messages
 */
const enTranslationMessages = require('translations/en.json');

/**
 * Getting and saving default locale for the project
 * You can edit in constants file only
 */
const { DEFAULT_LOCALE } = require('constants/locale');
localStorage.setItem('locale', DEFAULT_LOCALE);

const formatTranslationMessages = (locale, messages) => {
  const defaultFormattedMessages =
    locale !== DEFAULT_LOCALE
      ? formatTranslationMessages(DEFAULT_LOCALE, enTranslationMessages)
      : {};
  const flattenFormattedMessages = (formattedMessages, key) => {
    const formattedMessage =
      !messages[key] && locale !== DEFAULT_LOCALE
        ? defaultFormattedMessages[key]
        : messages[key];
    return Object.assign(formattedMessages, { [key]: formattedMessage });
  };
  return Object.keys(messages).reduce(flattenFormattedMessages, {});
};

export const translationMessages = {
  en: formatTranslationMessages('en', enTranslationMessages),
};
