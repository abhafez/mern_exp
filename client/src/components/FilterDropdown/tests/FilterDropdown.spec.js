import { render, screen, waitFor } from '@testing-library/react';
import { IntlProvider } from 'react-intl';
import { departments } from 'constants/filters';
import FilterDropdown from '../FilterDropdown';
import { translationMessages } from '../../../i18n';

describe('<FilterDropdown />', () => {
  it('Should Contain Options Data', () => {
    const onFilterSpy = jest.fn();
    const value = 'some value';
    const EN = 'en';
    const { container } = render(
      <IntlProvider locale={EN} messages={translationMessages[EN]}>
        <FilterDropdown value={value} options={departments} onFilter={onFilterSpy} />
      </IntlProvider>,
    );
    expect(container.firstChild).toMatchSnapshot();
    expect(onFilterSpy).not.toHaveBeenCalled();
    waitFor(() => expect(screen.getByTestId('dropdown')).toContain('Gaming'));
  });
});
