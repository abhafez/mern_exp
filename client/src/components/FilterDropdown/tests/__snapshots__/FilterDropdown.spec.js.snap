// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FilterDropdown /> Should Contain Options Data 1`] = `
<div
  aria-expanded="false"
  class="ui search selection dropdown"
  role="combobox"
  testid="dropdown"
>
  <input
    aria-autocomplete="list"
    autocomplete="off"
    class="search"
    tabindex="0"
    type="text"
    value=""
  />
  <div
    aria-atomic="true"
    aria-live="polite"
    class="text"
    role="alert"
  />
  <i
    aria-hidden="true"
    class="dropdown icon"
  />
  <div
    class="menu transition"
    role="listbox"
  >
    <div
      aria-checked="false"
      aria-selected="true"
      class="selected item"
      role="option"
      style="pointer-events: all;"
    >
      <span
        class="text"
      >
        Gaming
      </span>
    </div>
    <div
      aria-checked="false"
      aria-selected="false"
      class="item"
      role="option"
      style="pointer-events: all;"
    >
      <span
        class="text"
      >
        Toys
      </span>
    </div>
    <div
      aria-checked="false"
      aria-selected="false"
      class="item"
      role="option"
      style="pointer-events: all;"
    >
      <span
        class="text"
      >
        Gifts
      </span>
    </div>
    <div
      aria-checked="false"
      aria-selected="false"
      class="item"
      role="option"
      style="pointer-events: all;"
    >
      <span
        class="text"
      >
        Books
      </span>
    </div>
  </div>
</div>
`;
